src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,107 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int PRIME_NUMBER = 31;
+    private static final int PRIME_NUMBER1 = 311;
+    private int capacity;
+    private Node<K, V>[] table;
+    private int threshold;
+    private int modifications;
+    private int size;
 
     @Override
     public void put(K key, V value) {
-
+        if (size == 0 && table == null) {
+            capacity = DEFAULT_CAPACITY;
+            threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+            table = (Node<K, V>[]) new Node[DEFAULT_CAPACITY];
+        }
+        Node<K, V> newNode = new Node<>(hash(key), key, value, null);
+        int index = bucketIndex(key);
+        if (table[index] == null) {
+            table[index] = newNode;
+        } else if (table[index].key == key
+                || key != null && table[index].key.equals(key)) {
+            table[index].value = newNode.value;
+            size--;
+        } else if (table[index].next == null) {
+            table[index].next = newNode;
+        } else {
+            Node<K, V> node = table[index];
+            while (node.next != null) {
+                node = node.next;
+            }
+            node.next = newNode;
+        }
+        modifications++;
+        size++;
+        if (modifications == threshold) {
```

Comment:
You should compare your threshold with size instead of modifications, cause modifications always grow and I think  int modifications is redundant.